control-plane/agents/src/bin/ha/node/server.rs

```diff
@@ -116,9 +116,19 @@ async fn disconnect_controller(
                 });
             }
 
-            if subsystem
-                .address
-                .match_host_port(new_path_uri.host(), &new_path_uri.port().to_string())
+            let ctrlr_transport = subsystem
+                .transport
+                .parse()
+                .map_err(|_| SvcError::InvalidArguments {})?;
+            let ctrlr_xprt_same_as_new_path = new_path_uri.transport().eq(&ctrlr_transport);
+
+            // It is same controller only if transport also match. Otherwise we disconnect it.
+            // This also takes care to disconnect old path if transport hasn't changed but host addr
+            // changed, for example due to interface name modified in io-engine args.
+            if ctrlr_xprt_same_as_new_path
+                && subsystem
+                    .address
+                    .match_host_port(new_path_uri.host(), &new_path_uri.port().to_string())
```

Comment:
```suggestion
            let transport = subsystem
                .transport
                .parse()
                .map_err(|_| SvcError::InvalidArguments {})?;
            let same_transport = new_path_uri.transport().eq(&ctrlr_transport);
            let same_host_port = subsystem.address.match_host_port(new_path_uri.host(), &new_path_uri.port().to_string());

            // It is same controller only if the transport is also a match. Otherwise we disconnect it.
            // This also takes care to disconnect old path if the transport hasn't changed but host addr
            // changed, for example due to interface name modified in io-engine args.
            if same_transport && same_host_port
```

Reply:
Also consider putting this in a helper function, taking both the new_path_uri and the subsystem.

Reply:
> Also consider putting this in a helper function, taking both the new_path_uri and the subsystem.

done.